features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/presequite/CpuRequirementChecker.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.presequite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.listFilesSafe
+import java.io.File
+
+internal class CpuRequirementChecker(
+    private val cpuDirFile: File = File(DIR_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Int? = null
+
+    override fun checkMinimumRequirement(systemRequirementConfiguration: SystemRequirementConfiguration): Boolean {
+        return readCpuCoreNumber() >= systemRequirementConfiguration.cpuCoreNumber
+    }
+
+    override fun name(): String = CPU_CHECK_NAME
+
+    override fun checkedValue(): Any? = checkedValue
+
+    private fun readCpuCoreNumber(): Int {
+        val files = cpuDirFile.listFilesSafe(internalLogger) { _, name -> name.matches(REGEX_CPU_CORE_FILE) }
+        val value = files?.size ?: fallbackReadCpuCoreNumber()
+        checkedValue = value
+        return value
+    }
```

Comment:
this will trigger `StrictMode` violations because of the I/O on the main thread

Reply:
Good point, now I dispatch the `SessionReplayRequirement.=runIfMeetRequirements` function into another thread, and unit test is updated accordingly, please check it.